### PR TITLE
memprofile: add http debug test

### PR DESCRIPTION
**Pull request by christoph-zededa**
## Summary by CodeRabbit

- **New Features**
	- Introduced improved error handling for server operations.
	- Added a new test for memory pressure monitoring functionality.
  
- **Bug Fixes**
	- Enhanced error distinction during server shutdown to improve logging accuracy.

- **Refactor**
	- Renamed `listenDebug` to `ListenDebug` to increase its accessibility.
	- Changed `PressureMemoryFile` from a constant to a variable for runtime modifications.

### CODE REVIEW

File: pkg/pillar/agentlog/agentlog_test.go [hunk 1]
```diff
@@ -219,3 +223,79 @@ func TestPubsubLog(t *testing.T) {
...
+
+func TestMemoryPressure(t *testing.T) {
+	realPressureMemoryProfile := agentlog.PressureMemoryFile
+	defer func() {
+		agentlog.PressureMemoryFile = realPressureMemoryProfile
+	}()
+
+	mpfh, err := os.CreateTemp("", "memory-pressure")
+	if err != nil {
+		panic(err)
```
[the remark this answers is not in the document]
Comment (OhmSpectator) [reply pkg/pillar/agentlog/agentlog_test.go:235]: Do we really need these "better test output and integration with testing tools" in the scope of this test?

[the remark this answers is not in the document]
Comment (OhmSpectator) [reply pkg/pillar/agentlog/agentlog_test.go:235]: I think it makes sense to use both `panic` and `t.Fatalf` as this way, we can distinguish between the errors caused by generic system failure (testing env is broken - use `panic`) and real test failure (use `t.Fatalf` for that). Or are there better approaches?

File: pkg/pillar/agentlog/memprofile.go [hunk 2]
```diff
@@ -42,6 +40,9 @@ const (
 	StatLineSize = 75
 )
 
+// PressureMemoryFile is the memory pressure file
+var PressureMemoryFile = "/proc/pressure/memory"
```
Comment (OhmSpectator) [inline pkg/pillar/agentlog/memprofile.go:44]: @christoph-zededa, I'm not too fond of making such a value a variable. It's a constant. Is there a way to mock a constant in Go? Or maybe we can introduce conditional builds with different values for test env and prod env?

Comment (christoph-zededa) [reply pkg/pillar/agentlog/memprofile.go:44]: We could f.e. make the path a parameter to `listenDebug` and when we invoke it from the tests, we can give it a different paths than in the real system.

Comment (OhmSpectator) [reply pkg/pillar/agentlog/memprofile.go:44]: Yeah... It looks like a proper solution...

Comment (OhmSpectator) [reply pkg/pillar/agentlog/memprofile.go:44]: On the other hand, as I consider adding parameters to all the functions, I start thinking about the possibility of collecting PSI stats for not just memory but also CPU and IO. In that case, we could make the PSI collector more generic, move it to another file, or even a different component. However, I then realize that we don't actually need all of that right now, and I hesitate to redo all the code now. 

So, can we mock a constant at this point?

File: pkg/pillar/agentlog/agentlog_test.go [hunk 3]
```diff
@@ -219,3 +223,79 @@ func TestPubsubLog(t *testing.T) {
...
+	}
+
+	err = os.MkdirAll(types.MemoryMonitorOutputDir, 0755)
```
Comment (OhmSpectator) [inline pkg/pillar/agentlog/agentlog_test.go:238]: It will fail to create a directory at `/persist` on a host system...

**Comment by OhmSpectator** [comment]
I find the approach useful, but we still have to integrate it better...

File: pkg/pillar/agentlog/agentlog_test.go (hunk 3 above)
Comment (OhmSpectator) [reply pkg/pillar/agentlog/agentlog_test.go:238]: Or is it supposed to run only in a container? How do you start the tests? Not with `go test`?

Comment (christoph-zededa) [reply pkg/pillar/agentlog/agentlog_test.go:238]: `os.MkdirAll` is like `mkdir -p` it won't fail if the directory already exists.

I start the test either from an ubuntu container https://github.com/lf-edge/eve/compare/master...christoph-zededa:eve:rr or from the eve-build container or the pillar container itself.

`make test` runs all tests in a container - so it takes a while ...

**Comment by OhmSpectator** [comment]
@christoph-zededa, I wantd to push an extra commit on top of the PR, but got this
```
ERROR: Permission to christoph-zededa/eve.git denied to OhmSpectator.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by christoph-zededa** [comment]
> @christoph-zededa, I wantd to push an extra commit on top of the PR, but got this
> 
> ```
> ERROR: Permission to christoph-zededa/eve.git denied to OhmSpectator.
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
> ```

I guess you either have to create a PR into my branch which is a PR into your branch.
Or you merge/cherry-pick my branch into your branch and put a commit on top.

**Comment by OhmSpectator** [comment]
> I guess you either have to create a PR into my branch which is a PR into your branch.
Or you merge/cherry-pick my branch into your branch and put a commit on top.

I wanted to draft the approach in this PR. 
To push changes into the PR, I should have collaborator permissions for your EVE fork, as I did for you in my fork (so you can push changes into my PRs).

File: pkg/pillar/agentlog/agentlog_test.go (hunk 3 above)
Comment (OhmSpectator) [reply pkg/pillar/agentlog/agentlog_test.go:238]: Aha... In the pillar dir, I can run `make build-docker-env enter-docker-env`, and the tests work there. It's not very convenient, but it works.

**Pull request closed without merging**